plugins/in_opentelemetry/opentelemetry_prot.c

```diff
@@ -2435,19 +2436,37 @@ int opentelemetry_prot_handle_ng(struct flb_http_request *request,
         strcmp(request->path, "/opentelemetry.proto.collector.metric.v1.MetricService/Export") == 0 ||
         strcmp(request->path, "/opentelemetry.proto.collector.metrics.v1.MetricsService/Export") == 0) {
         payload_type = 'M';
-        result = process_payload_metrics_ng(context, context->ins->tag, request, response);
+        if (context->tag_from_uri == FLB_TRUE) {
+            tag = flb_sds_create("v1_metrics");
+        }
+        else {
+            tag = flb_sds_create(context->ins->tag);
+        }
+        result = process_payload_metrics_ng(context, tag, request, response);
     }
     else if (strcmp(request->path, "/v1/traces") == 0 ||
              strcmp(request->path, "/opentelemetry.proto.collector.trace.v1.TraceService/Export") == 0 ||
              strcmp(request->path, "/opentelemetry.proto.collector.traces.v1.TracesService/Export") == 0) {
         payload_type = 'T';
-        result = process_payload_traces_ng(context, context->ins->tag, request, response);
+        if (context->tag_from_uri == FLB_TRUE) {
+            tag = flb_sds_create("v1_traces");
+        }
+        else {
+            tag = flb_sds_create(context->ins->tag);
+        }
+        result = process_payload_traces_ng(context, tag, request, response);
     }
     else if (strcmp(request->path, "/v1/logs") == 0 ||
              strcmp(request->path, "/opentelemetry.proto.collector.log.v1.LogService/Export") == 0 ||
              strcmp(request->path, "/opentelemetry.proto.collector.logs.v1.LogsService/Export") == 0) {
         payload_type = 'L';
-        result = process_payload_logs_ng(context, context->ins->tag, request, response);
+        if (context->tag_from_uri == FLB_TRUE) {
+            tag = flb_sds_create("v1_logs");
+        }
+        else {
+            tag = flb_sds_create(context->ins->tag);
```

Comment:
if I am not wrong, this new memory buffer is never released, please validate that on every place is created.

Reply:
Oh yeah, good catch. Although, the tag should get handed off to the routing/processing system later on via the `flb_input_log_append`/`flb_input_metrics_append`/`flb_input_trace_append` calls - e.g. https://github.com/fluent/fluent-bit/blob/69bf966c5f073d116a51f2b4d00590ae5367d13b/plugins/in_opentelemetry/opentelemetry_prot.c#L2389-L2395

Does this mean the buffer gets picked up by another thread later? (does it need to be freed there? does that perhaps already happen?)

Reply:
Hmm, looking at the existing HTTP1 codepath, it seems the tag does get freed there by the handler at the end:

https://github.com/fluent/fluent-bit/blob/69bf966c5f073d116a51f2b4d00590ae5367d13b/plugins/in_opentelemetry/opentelemetry_prot.c#L1923-L1928